gspread/utils.py

```diff
@@ -27,6 +27,8 @@
 URL_KEY_V1_RE = re.compile(r"key=([^&#]+)")
 URL_KEY_V2_RE = re.compile(r"/spreadsheets/d/([a-zA-Z0-9-_]+)")
 
+TITLE_RANGE_RE = re.compile(r"'(.*?)'!.*")
```

Comment:
This won't always work, here we can have 2 cases:
1. names with a ` ` blank space or that starts with `'` so they start with a `'` but they may have multiple `'` surrounding the actual name
2. no ` ` blank space, no `'` at starts or end. so it's a single string with only characters.

The regex should be improved to match all possible titles.